src/main/java/com/celements/pagelayout/LayoutScriptService.java

```diff
@@ -49,54 +51,101 @@
+  private LayoutServiceRole layoutService;
+
+  @Requirement
+  private ModelContext modelContext;
+
+  @Requirement
+  private ModelUtils modelUtils;
+
+  @Requirement
+  private DefaultRightsAccessFacade rightsAccess;
 
   private XWikiContext getContext() {
-    return (XWikiContext) execution.getContext().getProperty("xwikicontext");
+    return modelContext.getXWikiContext();
   }
 
   public String renderPageLayout() {
-    return getPageLayoutCmd().renderPageLayout();
+    return layoutService.renderPageLayout();
   }
 
-  public String renderPageLayout(SpaceReference spaceRef) {
-    return getPageLayoutCmd().renderPageLayout(spaceRef);
+  @NotNull
+  public String renderPageLayout(@Nullable SpaceReference spaceRef) {
+    return layoutService.renderPageLayout(spaceRef);
   }
 
+  /**
+   * @deprecated since 5.4 instead use {@link #getActivePageLayoutSpaceRefs()}
+   */
+  @Deprecated
   public Map<String, String> getActivePageLayouts() {
     return getPageLayoutCmd().getActivePageLyouts();
   }
 
+  /**
+   * @deprecated since 5.4 instead use {@link #getAllPageLayouts()}
+   */
+  @Deprecated
   public Map<String, String> getAllPageLayouts() {
     return getPageLayoutCmd().getAllPageLayouts();
   }
 
-  public String createNewLayout(String layoutSpaceName) {
-    return getPageLayoutCmd().createNew(getWebUtilsService().resolveSpaceReference(
-        layoutSpaceName));
-  }
-
-  public boolean deleteLayout(String layoutSpaceName) {
-    SpaceReference layoutSpaceRef = getWebUtilsService().resolveSpaceReference(layoutSpaceName);
-    String layoutPropDocName = getEntitySerializer().serialize(
-        getPageLayoutCmd().standardPropDocRef(layoutSpaceRef));
-    try {
-      if (getContext().getWiki().getRightService().hasAccessLevel("delete", getContext().getUser(),
-          layoutPropDocName, getContext())) {
-        return getPageLayoutCmd().deleteLayout(layoutSpaceRef);
-      } else {
-        LOGGER.warn("NO delete rights on [" + layoutPropDocName + "] for user ["
-            + getContext().getUser() + "].");
-      }
-    } catch (XWikiException exp) {
-      LOGGER.error("Failed to check delete rights on [" + layoutSpaceName + "] for user ["
-          + getContext().getUser() + "].");
+  @NotNull
+  public Map<SpaceReference, String> getActivePageLayoutSpaceRefs() {
+    return layoutService.getActivePageLayouts();
+  }
+
+  @NotNull
+  public Map<SpaceReference, String> getAllPageLayoutSpaceRefs() {
+    return layoutService.getAllPageLayouts();
+  }
+
+  /**
+   * @deprecated since 5.4 instead use {@link #createLayout(SpaceReference)}
+   */
+  @Deprecated
+  @NotNull
+  public String createNewLayout(@Nullable String layoutSpaceName) {
+    SpaceReference layoutSpaceRef = Optional
+        .ofNullable(modelUtils.resolveRef(layoutSpaceName, SpaceReference.class)).orElse(null);
+    return createLayout(layoutSpaceRef);
+  }
+
+  @NotNull
+  public String createLayout(@Nullable SpaceReference layoutSpaceRef) {
```

Comment:
missing rights checks

Reply:
never had rights check. createLayout is no rightful operation